### PR TITLE
ci: switch to github actions, fix some linting issues

### PR DESCRIPTION
**Pull request by domoritz**
Faster and more reliable builds

### CODE REVIEW

**Comment by ctb** [comment]
yay!

**Comment by domoritz** [comment]
The setup seems fine but tests are failing. Do you know why?

Screen Shot 2020-10-20 at 16 36 58

**Comment by ctb** [comment]
Looks like #325 fixes it, yay! Adds a `pip install -e`.

Test run --

https://github.com/spacegraphcats/spacegraphcats/runs/1287399406?check_suite_focus=true

code cov results!!

https://codecov.io/gh/spacegraphcats/spacegraphcats/tree/1bd8d18207b6be57a3a99df3b96f63298a104114

File: Makefile [hunk 1]
```diff
@@ -1,13 +1,13 @@
 all: test
 
 flakes:
-	pyflakes search/*.py
+	pyflakes spacegraphcats/*.py
```
Comment (ctb) [inline Makefile]: note, this only looks at spacegraphcats/*.py, not the stuff under that directory. Would probably need to do
```suggestion
 	pyflakes $(find spacegraphcats -name \*.py)
```

**Review by ctb — APPROVED** [review]
thank you!

File: Makefile (hunk 1 above)
Comment (domoritz) [reply Makefile]: Looks like `pyflakes spacegraphcats` works as well.

**Review by ctb — APPROVED** [review]

**Comment by domoritz** [comment]
Btw, there are quite a few unused variables but I didn't fix them in this pull request as I wanted to keep it somewhat small. See https://github.com/spacegraphcats/spacegraphcats/pull/323/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R4

**Comment by ctb** [comment]
thanks! will dig in after this gets merged!

On Wed, Oct 21, 2020 at 09:43:04AM -0700, Dominik Moritz wrote:
> Btw, there are quite a few unused variables but I didn't fix them in this pull request as I wanted to keep it somewhat small. See https://github.com/spacegraphcats/spacegraphcats/pull/323/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R4
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/spacegraphcats/spacegraphcats/pull/323#issuecomment-713707982
-- 
C. Titus Brown, ctbrown@ucdavis.edu

**Pull request merged**